### DISCUSSION TITLE
Generate a companion object for the enums

### DISCUSSION THREAD

**Discussion by ghost** [Netflix/dgs-framework#673] [Ideas]
Hi everyone a quick and short idea. I'm seeing that every object that is created and generated automatically based on the schema contains a companion object, for me this is really useful because i tend to write the mappers inside the companion objects like this:

data class ExampleType{
....
companion object
}

fun ExampleType.Companion.from(){
//whatever mapping i want to do between domain models and my graphql models
}

The problem I have is that the enums doesn't have a companion object so i cannot write an extension function.

Thanks!

**Comment by srinivasankavitha** [comment]
That sounds useful. Will add that. Could you create a github issue here with the details above and an example: https://github.com/Netflix/dgs-codegen/issues

**Reply by ghost** [reply]
I'll do that, thank you!!

**Reply by shapan1** [reply]
@zanonena @srinivasankavitha Can you please point me to how to have a companion object generated from graphql using codegen?
